### PR TITLE
IntakeStates ➝ subsystems-dev

### PR DESCRIPTION
**Pull request by maya1414**

### CODE REVIEW

**Review by hillelv — COMMENTED** [review]
move base branch to states-dev and add the handler to the superstructure

File: src/main/java/frc/robot/subsystems/intake/IntakeStates.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package frc.robot.subsystems.intake;
+
+public enum IntakeStates {
+
+	NOTE_TO_SHOOTER(0),
+	INTAKE(0),
+	STOP(0),
+	SHOOTER_TO_ELEVATOR(0);
```
Comment (drorta) [inline src/main/java/frc/robot/subsystems/intake/IntakeStates.java]: elevator to shooter

File: src/main/java/frc/robot/subsystems/intake/roller/IntakeStates.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package frc.robot.subsystems.intake.roller;
+
+public enum IntakeStates {
+
+	NOTE_TO_SHOOTER(0),
+	INTAKE(0),
+	STOP(0),
+	ELEVATOR_TO_SHOOTER(0);
```
Comment (Yotam7) [inline src/main/java/frc/robot/subsystems/intake/roller/IntakeStates.java:8]: make `STOP` state last
```suggestion
	ELEVATOR_TO_SHOOTER(0),
	STOP(0);
```

Comment (Yotam7) [inline src/main/java/frc/robot/subsystems/intake/roller/IntakeStates.java]: change order of `IntakeStates` to be more inline with what happens in games
```suggestion
	INTAKE(0),
	NOTE_TO_SHOOTER(0),
	ELEVATOR_TO_SHOOTER(0),
	STOP(0);
```

Comment (hillelv) [inline src/main/java/frc/robot/subsystems/intake/roller/IntakeStates.java]: shooter to elevator

File: src/main/java/frc/robot/subsystems/intake/roller/IntakeStates.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package frc.robot.subsystems.intake.roller;
+
+public enum IntakeStates {
```
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/intake/roller/IntakeStates.java:3]: add outtake state

File: src/main/java/frc/robot/superstructure/Superstructure.java [hunk 4]
```diff
@@ -100,28 +108,32 @@ public Command amp() {
...
+			intakeStatesHandler.setState(IntakeStates.NOTE_TO_SHOOTER)
 		);
 	}
 
 	public Command intakeOuttake() {
 		return new ParallelCommandGroup(
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
-			funnelStateHandler.setState(FunnelState.INTAKE_OUTTAKE)
+			funnelStateHandler.setState(FunnelState.INTAKE_OUTTAKE),
+			intakeStatesHandler.setState(IntakeStates.STOP)
```
Comment (YHerm) [inline src/main/java/frc/robot/superstructure/Superstructure.java]: should do outtake

**Review by Yotam7 — DISMISSED** [review]

**Review by YoniKiriaty — DISMISSED** [review]

**Review by YoniKiriaty — DISMISSED** [review]

**Review by YHerm — DISMISSED** [review]

**Review by Yotam7 — DISMISSED** [review]

File: src/main/java/frc/robot/subsystems/intake/roller/Calibrations [hunk 5]
```diff
@@ -2,4 +2,9 @@ Calibration:
 
 1. Gear Ratio
 2. Current Limit
-3. Motor id
\ No newline at end of file
+3. Motor id
+4. NOTE_TO_SHOOTER power
+5. INTAKE power
+6. STOP power
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/subsystems/intake/roller/Calibrations]: Why do you need to calibrate stop power..?

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Review by Yotam7 — APPROVED** [review]

**Review by YHerm — APPROVED** [review]

**Review by YoniKiriaty — APPROVED** [review]

**Review by drorta — APPROVED** [review]

**Review by hillelv — APPROVED** [review]

**Pull request merged**